### PR TITLE
add `oncall_review_assigner` step into `issue-opened.yml` workflow

### PR DESCRIPTION
**Pull request by BBBmau**
### Description

This PR is for adding a new job into our issue/pr triaging workflow where it assigns the opened issue/PR to the appropriate bug on duty (based off the response from the pagerduty api)

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: .github/workflows/review-assigner.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                 if (context.eventName == "issues"){
+                     try{
+                         await github.rest.issues.addAssignees({
+                             owner: context.repo.owner,
+                             repo: context.repo.repo,
+                             assignees: ["BBBmau"], // this will be replaced with pagerduty resp
+                             issue_number: context.issue.number, // issue.number is used for both issues and PRs
+                         })
+                     } catch(error){
+                         core.setFailed("error assigning assignee:" + error)
```
Comment (alexsomesan) [inline .github/workflows/review-assigner.yaml:27]: The repetition sounds funny. Maybe rephrase with "reviewer" ?
```suggestion
                         core.setFailed("error assigning reviewer:" + error)
```

**Comment by BBBmau** [comment]
Some runs of this new GHA step can be seen here where its ran as its own GHA workflow: https://github.com/BBBmau/terraform-provider-kubernetes/actions/workflows/review-assigner.yaml

reviewer assigned can be done in a separate PR since me and @sheneska ran into some permission issues. Works fine with issue assigning.

**Comment by BBBmau** [comment]
Marking as Ready for Review now that `PAGERDUTY_TOKEN` is added as part of the repo environment variables

**Review by sheneska — APPROVED** [review]
Thanks for the changes! LGTM 🚀
